### PR TITLE
FEATURE: Add support for readonly classes

### PR DESCRIPTION
**Pull request by robertlemke**
Flow now respects readonly classes during proxy class building and makes sure that proxy classes are readonly as well.

resolves: #3025

### CODE REVIEW

**Comment by robertlemke** [comment]
Note that this PR is based on https://github.com/neos/flow-development-collection/pull/3050 which needs to be merged first.

Diff: https://github.com/robertlemke/flow-development-collection/compare/3049-only-add-constructor-injection-code-if-needed...robertlemke:flow-development-collection:3025-add-support-for-readonly-classes

File: Neos.Flow/Tests/Functional/ObjectManagement/Fixtures/ValueObjectClassA.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * source code.
+ */
+
+/**
+ * A readonly class in the style of a value object
+ */
+readonly class ValueObjectClassA implements \JsonSerializable
+{
+    public function __construct(
+        readonly public string $value,
```
Comment (mhsdesign) [inline Neos.Flow/Tests/Functional/ObjectManagement/Fixtures/ValueObjectClassA.php:20]: sorry nitpicky: i think we can drop the `readonly` in the property then

**Review by kdambekalns — APPROVED** [review]
Looks good!

File: Neos.Flow/Tests/Functional/ObjectManagement/Fixtures/ValueObjectClassA.php (hunk 1 above)
Comment (kdambekalns) [reply Neos.Flow/Tests/Functional/ObjectManagement/Fixtures/ValueObjectClassA.php:20]: It's to make sure it's really, really, really readonly! 😇

Comment (kdambekalns) [reply Neos.Flow/Tests/Functional/ObjectManagement/Fixtures/ValueObjectClassA.php:20]: Honestly, yes – but I'd mix it, and leave one in. Just to have yet another edge-case…

File: Neos.Flow/Classes/ObjectManagement/Proxy/ProxyClass.php [hunk 2]
```diff
@@ -216,6 +214,9 @@ public function render()
         } elseif ($this->reflectionService->isClassFinal($this->fullOriginalClassName)) {
             $classModifier = 'final ';
         }
+        if ($this->reflectionService->isClassReadonly($this->fullOriginalClassName)) {
+            $classModifier = 'readonly ';
```
Comment (kitsunet) [inline Neos.Flow/Classes/ObjectManagement/Proxy/ProxyClass.php]: wouldn't that override the `final` modifiers, if both existed?

Comment (kdambekalns) [reply Neos.Flow/Classes/ObjectManagement/Proxy/ProxyClass.php]: Good point. And would `abstract readonly` be a thing?

Comment (kitsunet) [reply Neos.Flow/Classes/ObjectManagement/Proxy/ProxyClass.php]: both are allowed, yes

Comment (kitsunet) [reply Neos.Flow/Classes/ObjectManagement/Proxy/ProxyClass.php]:
```suggestion
            $classModifier .= 'readonly ';
```

**Review by kitsunet — APPROVED** [review]
